shared/js/background/tracker-utils.js

```diff
@@ -198,6 +198,19 @@ function truncateReferrer (referrer, target) {
     return modifiedReferrer
 }
 
+/**
+ * Checks if a tracker is a first party by checking entity data
+ * @param {string} trackerUrl
+ * @param {string} siteUrl
+ * @param {object} request
+ * @returns {boolean}
+ */
+function isFirstPartyByEntity (trackerUrl, siteUrl, request) {
+    if (!trackerUrl) return false
+    const trackerData = trackers.getTrackerData(trackerUrl, siteUrl, request)
```

Comment:
I looked a bit into this function today and it's a bit scary and not really meant for what we are trying to do here (it does rule matching, CNAME resolution, etc. that I don't think we want/need here?). Could we use `trackers.findTrackerOwner` instead?

Reply:
Ah wait, we may want CNAME resolution (but probably not rule matching).